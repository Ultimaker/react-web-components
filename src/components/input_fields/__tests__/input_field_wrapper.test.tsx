// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import InputFieldWrapper, { InputFieldWrapperProps } from '../input_field_wrapper';

describe('The input field wrapper component', () => {
    let props: InputFieldWrapperProps;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testInputId',
            staticField: false,
            className: 'test-input-class',
            centerInputField: true,
            label: 'Test Input Label',
            labelLayoutWidth: '1/2',
            labelWidthBreakpoint: 'xs',
            infoLinkURL: 'http://info.url',
            infoText: 'Click for more info',
            preLabelElement: '*',
            required: true,
            submitted: false,
            validationError: 'A validation error!',
            showValidationError: true,
            touched: true,
            children: <div>child</div>,
            inputChildren: <div>input child</div>,
        };
        wrapper = shallow(<InputFieldWrapper {...props} />);
    });

    it('should render a wrapper with all options', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a static field', () => {
        wrapper.setProps({ required: false, staticField: true, touched: false });
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a wrapper without any options', () => {
        props = {
            id: 'testInputId',
            touched: true,
            children: null,
            inputChildren: null,
        };
        wrapper = shallow(<InputFieldWrapper {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a wrapper without the validation error', () => {
        wrapper.setProps({ showValidationError: false });
        expect(wrapper).toMatchSnapshot();
    });
});
