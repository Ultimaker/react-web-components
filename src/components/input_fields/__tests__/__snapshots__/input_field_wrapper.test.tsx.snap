// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The input field wrapper component should render a static field 1`] = `
<div
  className="input-field layout test-input-class hide-input"
>
  <InputFieldLabel
    id="testInputId"
    infoLinkURL="http://info.url"
    infoText="Click for more info"
    label="Test Input Label"
    labelLayoutWidth="1/2"
    labelWidthBreakpoint="xs"
    preLabelElement="*"
  />
  <div
    className="layout__item u-fit layout__item--middle test-input-class text-center"
  >
    <div
      className="static-field"
    >
      <div>
        child
      </div>
    </div>
  </div>
  <div
    className="layout__item u-fit input-children"
  >
    <div>
      input child
    </div>
  </div>
</div>
`;

exports[`The input field wrapper component should render a wrapper with all options 1`] = `
<div
  className="input-field layout test-input-class"
>
  <InputFieldLabel
    id="testInputId"
    infoLinkURL="http://info.url"
    infoText="Click for more info"
    label="Test Input Label"
    labelLayoutWidth="1/2"
    labelWidthBreakpoint="xs"
    preLabelElement="*"
  />
  <div
    className="layout__item u-fill layout__item--middle test-input-class text-center"
  >
    <div
      className="input-container layout layout--gutter-xs"
    >
      <div
        className="input-wrapper-children layout__item u-fill"
      >
        <div>
          child
        </div>
      </div>
      <div
        className="layout__item u-fit input-field__field-addition"
      >
        <Icon(RequiredIcon) />
      </div>
    </div>
  </div>
  <div
    className="layout__item u-fit input-children"
  >
    <div>
      input child
    </div>
  </div>
  <InputFieldValidation
    labelLayoutWidth="1/2"
    labelWidthBreakpoint="xs"
    required={true}
    validationError="A validation error!"
  />
</div>
`;

exports[`The input field wrapper component should render a wrapper without any options 1`] = `
<div
  className="input-field layout"
>
  <div
    className="layout__item u-fill layout__item--middle"
  >
    <div
      className="input-container layout layout--gutter-xs"
    >
      <div
        className="input-wrapper-children layout__item u-fill"
      />
    </div>
  </div>
</div>
`;

exports[`The input field wrapper component should render a wrapper without the validation error 1`] = `
<div
  className="input-field layout test-input-class"
>
  <InputFieldLabel
    id="testInputId"
    infoLinkURL="http://info.url"
    infoText="Click for more info"
    label="Test Input Label"
    labelLayoutWidth="1/2"
    labelWidthBreakpoint="xs"
    preLabelElement="*"
  />
  <div
    className="layout__item u-fill layout__item--middle test-input-class text-center"
  >
    <div
      className="input-container layout layout--gutter-xs"
    >
      <div
        className="input-wrapper-children layout__item u-fill"
      >
        <div>
          child
        </div>
      </div>
      <div
        className="layout__item u-fit input-field__field-addition"
      >
        <Icon(RequiredIcon) />
      </div>
    </div>
  </div>
  <div
    className="layout__item u-fit input-children"
  >
    <div>
      input child
    </div>
  </div>
</div>
`;
